calico_containers/tests/st/test_base.py

```diff
@@ -44,3 +48,32 @@ def start_etcd(self):
             initial_cluster="calico=http://%s:2380" % self.ip,
             initial_cluster_state="new",
         )
+
+    def assert_powerstrip_up(self, host):
+        powerstrip = partial(host.execute, "docker ps", docker_host=True)
+        self.retry_until_success(powerstrip, ex_class=ErrorReturnCode)
+
+    def retry_until_success(self, function, retries=10, ex_class=None):
+        """
+        Retries function until no exception is thrown. If exception continues,
+        it is reraised.
+
+        :param function: the function to be repeatedly called
+        :param retries: the maximum number of times to retry the function.
+        A value of 0 will run the function once with no retries.
+        :param ex_class: The class of expected exceptions.
+        :returns: the value returned by function
+        """
+        for retry in range(retries + 1):
+            try:
+                result = function()
+            except Exception as e:
```

Comment:
I feel like it would be cleaner to default `ex_class=Exception` and then on this line do `except ex_class as e:`.  That would remove the need for the `if not issubclass` statement.